nitransforms/base.py

```diff
@@ -251,14 +252,57 @@ class TransformBase:
     __slots__ = (
         "_reference",
         "_ndim",
+        "_affine",
+        "_shape",
+        "_header",
+        "_grid",
+        "_mapping",
+        "_hdf5_dct",
+        "_x5_dct",
     )
 
-    def __init__(self, reference=None):
+    x5_struct = {
+        "TransformGroup/0": {
+            "Type": None,
+            "Transform": None,
+            "Metadata": None,
+            "Inverse": None,
+        },
+        "TransformGroup/0/Domain": {"Grid": None, "Size": None, "Mapping": None},
+        "TransformGroup/1": {},
+        "TransformChain": {},
+    }
```

Comment:
We probably want to make this struct some sort of config option. Is there such a concept in nibabel, @effigies?

Reply:
I don't think so, but what kind of config option are you thinking? I'm very wary of putting mutable global state in a library, as that just asks for nasty interactions with multiple libraries (and threads). We could think about making a context manager to update state, along with [ContextVar](https://docs.python.org/3/library/contextvars.html)s to manage state, if needed.

But possibly you're talking about something else? This looks like a place for a schema, but I haven't studied this code yet.